src/renderer_vk.cpp

```diff
@@ -1203,6 +1204,21 @@ VK_IMPORT
 
 #undef VK_IMPORT_FUNC
 
+#else // VK_NO_PROTOTYPES
+
+ 			vkGetInstanceProcAddr = ::vkGetInstanceProcAddr;
+			vkGetDeviceProcAddr = ::vkGetDeviceProcAddr;
+#define VK_IMPORT_FUNC(_optional, _func)                      \
```

Comment:
Can you explain what's going on here?

Reply:
When VK_NO_PROTOTYPES is _not_ defined, Vulkan can be treated as any other library that is linked into our binary. To be able to get the address of all Vulkan functions, `vkGetInstanceProcAddr` is used (https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkGetInstanceProcAddr.html).

However obviously we can't use `vkGetInstanceProcAddr` to get the address of `vkGetInstanceProcAddr` itself, so for that function (and its twin, `vkGetDeviceProcAddr`), we simply take the address of the global function declared in the Vulkan headers.

I believe the VK_NO_PROTOTYPES path could also be changed to have a similar system (e.g. only find these two symbols in the dll file and then look up the rest of the function pointers using `vkGetInstanceProcAddr`), but I'm not 100% sure, and I'd rather not risk a change that is hard to test (due to many different platforms being affected and random computer setup differences).

It is probably also possible to take the same code path not only on NX but on any platform, by linking the Vulkan library with bgfx directly.

Reply:
> But it is always defined:

This pull request changes the code you just paster so that `VK_NO_PROTOTYPES` is not defined.

I'm not sure if I can share any details but I can certainly point out that in the bx repo `bx::dlopen`, which the currently implemented solution uses, is unimplemented on the NX platform.

Reply:
Point I'm trying to make is to not create more code paths that I can't test. Ideally you would just implement bx::dlopen, and the rest wouldn't change. See if some open source project like https://github.com/devkitPro/deko3d or accompanying libraries offers solution for dlopen.

Reply:
Thank you. At this point I'd rather not look at other implementations of get into a discussion of the platform to avoid legal issues :)
I fully understand your concern and if you don't want to merge this, that's fine. I can remove this part and keep the rest, more straightforward snippets.

Reply:
I deleted this snippet.